shared/chat/conversation/messages/react-button/delay-interval.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+// The first emoji should stay on screen for less time so the user has a better chance of seeing it.
+// Set the interval after a shorter initial delay.
+// Convenience wrapper around interval + timeout
+class DelayInterval {
+  _intervalMS: number
+  _delayMS: number
+
+  _intervalID: IntervalID
+  _delayID: TimeoutID
+
+  running: boolean
+
+  constructor(intervalMS: number, delayMS: number) {
+    this._intervalMS = intervalMS
+    this._delayMS = delayMS
+  }
+  start(fcn: () => void) {
+    if (this.running) {
```

Comment:
should remove running and just have that be a function of _delayID || _intervalID